src/foam/nanos/approval/ApprovableApprovalRequestsRule.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.approval',
+  name: 'ApprovableApprovalRequestsRule',
+
+  documentation: `
+    A rule to update the approvable once it's related approval request has been
+    APPROVED or REJECTED
+  `,
+
+  javaImports: [
+    'foam.core.ContextAwareAgent',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.approval.Approvable',
+    'foam.nanos.approval.ApprovalRequest',
+    'foam.nanos.approval.ApprovalStatus',
+    'foam.nanos.ruler.Operations',
+  ],
+
+  implements: ['foam.nanos.ruler.RuleAction'],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        ApprovalRequest request = (ApprovalRequest) obj.fclone();
+
+        agency.submit(x, new ContextAwareAgent() {
+          
+          @Override
+          public void execute(X x) {
+            DAO approvableDAO = (DAO) getX().get("approvableDAO");
```

Comment:
is approvableDAO the Approvable.daoKey?

Reply:
got it, see it below.

Reply:
No its the actual approvableDAO where all the approvables are stored.